File: frontend/src/microservices/operations/objects/ManagerRefundTaskListResponse.ts
// 本文件定义 operations 模块的 `ManagerRefundTaskListResponse`，作为列表响应数据并提供 JSON 编解码。

import type { ManagerRefundTaskResponse } from './ManagerRefundTaskResponse'

export type ManagerRefundTaskListResponse = {
  tasks: ManagerRefundTaskResponse[]
}
export const managerRefundTaskListResponseFromJson = (json: string): ManagerRefundTaskListResponse =>
  JSON.parse(json) as ManagerRefundTaskListResponse

export const managerRefundTaskListResponseToJson = (value: ManagerRefundTaskListResponse): string =>
  JSON.stringify(value)

File: backend/src/main/scala/microservices/operations/overall/objects/ManagerRefundTaskListResponse.scala
// 本文件定义 `ListManagerRefundTasksPlanner` 返回的退款任务列表对象。
package com.typesafe.travel.operations.domain

import io.circe.{Decoder, Encoder}
import io.circe.generic.semiauto.{deriveDecoder, deriveEncoder}

final case class ManagerRefundTaskListResponse(tasks: List[ManagerRefundTaskResponse])
object ManagerRefundTaskListResponse:
  given sourceEncoder: Encoder[ManagerRefundTaskListResponse] = deriveEncoder
  given sourceDecoder: Decoder[ManagerRefundTaskListResponse] = deriveDecoder
